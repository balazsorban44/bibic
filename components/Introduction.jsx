import React from 'react'

const profiles = [
  {
    name: "Hegedüs Péter",
    src: "/images/intro/hegedus.jpg",
    email: "hegedus@bibicvendeghazak.hu",
    position: "Bíbic Vendégházak házigazdája"
  },
  {
    name: "Hegedüsné Kóró Ágnes",
    src: "/images/intro/hegedusne.jpg",
    email: "hegedusne@bibicvendeghazak.hu",
    position: "Tulajdonos"
  },
  {
    name: "Gombkötő Gábor",
    src: "/images/intro/gombkoto.jpg",
    email: "gombkoto@bibicvendeghazak.hu",
    position: "Tulajdonos"
  }
]

const Introduction = ({paragraphs}) =>
  <section id="bemutatkozas">
    <h2 className="welcome-title">Üdvözöljük</h2>
    <ul className="profiles">
      {profiles.map(({
        name, email, src, position
      }) =>
        <li key={name}>
          <a href={`mailto:${email}`}>
            <img
              alt={name}
              className="profile-img"
              src={src}
            />
          </a>
          <div>
            <h3>{name}</h3>
            <h5><a href={`mailto:${email}`}>✉ {email} </a></h5>
            <h4>{position}</h4>
          </div>
        </li>
      )}
    </ul>
    <div className="history">
      {paragraphs.map(({text}, index) =>
        <p key={index}>{text}</p>
      )}
    </div>
  </section>

export default Introduction