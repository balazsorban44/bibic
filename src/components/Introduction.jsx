import React, {Component} from 'react'
import hegedus from '../assets/images/intro/hegedus.jpg'
import hegedusne from '../assets/images/intro/hegedusne.jpg'
import gombkoto from '../assets/images/intro/gombkoto.jpg'
import Paragraphs from './shared/Paragraphs'


const profiles = [
  {
    name: "Hegedüs Péter",
    src: hegedus,
    email: "hegedus@bibicvendeghazak.hu",
    position: "Bíbic Vendégházak házigazdája"
  },
  {
    name: "Hegedüsné Kóró Ágnes",
    src: hegedusne,
    email: "hegedusne@bibicvendeghazak.hu",
    position: "Tulajdonos"
  },
  {
    name: "Gombkötő Gábor",
    src: gombkoto,
    email: "gombkoto@bibicvendeghazak.hu",
    position: "Tulajdonos"
  }
]

export default class Introduction extends Component {
  render() {
    return (
      <section id="bemutatkozas">
        <h2 className="welcome-title">Üdvözöljük</h2>
        <ul className="profiles">
          {profiles.map(({
            name, email, src, position
          }) =>
            <li key={name}>
              <a href={`mailto:${email}`}>
                <img
                  alt={name}
                  className="profile-img"
                  src={src}
                />
              </a>
              <div>
                <h3>{name}</h3>
                <h5><a href={`mailto:${email}`}>✉ {email} </a></h5>
                <h4>{position}</h4>
              </div>
            </li>
          )}
        </ul>
        <div className="history">
          <Paragraphs path="bemutatkozas"/>
        </div>
      </section>
    )
  }
}