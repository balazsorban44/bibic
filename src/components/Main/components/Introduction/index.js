import React from 'react'

import hegedus from '../../../../media/images/other/hegedus.jpg'
import hegedusne from '../../../../media/images/other/hegedusne.jpg'
import gombkoto from '../../../../media/images/other/gombkoto.jpg'

const Introduction = () => (
  <section id="bemutatkozas">
    <ul className="profiles">
      <li>
        <img className="profile-img" src={hegedus} alt="Hegedüs Péter"/>
        <div>
          <h3>Hegedüs Péter</h3>
          <h4>Bíbic Vendégházak házigazdája</h4>
        </div>
      </li>
      <li>
        <img className="profile-img" src={hegedusne} alt="Hegedüsné Kóró Ágnes"/>
        <div>
          <h3>Hegedüsné Kóró Ágnes</h3>
          <h4>Fauna ház tulajdonosa</h4>
        </div>
      </li>
      <li>
        <img className="profile-img" src={gombkoto} alt="Gombkötő Gábor"/>
        <div>
          <h3>Gombkötő Gábor</h3>
          <h4>Flóra ház tulajdonosa</h4>
        </div>
      </li>
    </ul>
    <div className="history">
      <p>A nagybajomi panzió, illetve vendégház tervét családi ötletelés - <b>Hegedüs Péter</b> és <b>Gombkötő Gábor</b>, testvérek vagyunk - alkalmával találtuk ki <b>1998</b>-ban.</p>
      <p>Hosszú évekig csak ötlet, illetve tervezgetés maradt a finanszírozás hiánya miatt. 2008-ban azonban a <b>Mezőgazdasági és Vidékfejlesztési Hivatal pályázat</b>ot írt ki, turisztikai fejlesztések támogatására.</p>
      <p>Ekkor, <b>két négy napraforgós vendégház</b> építésére támogatási kérelmet nyújtottunk be - Hegedüsné Kóró Ágnes és Gombkötő Gábor - a Hivatalhoz, melyet 2009-ben meg is ítéltek számunkra. Az előkészületek és az előfinanszírozás megszervezése után 2010-ben elkezdődhetett a <a href="#szobak">Flóra ház </a>építése. <b>2012-ben kaptunk működési engedélyt.</b> Jelenleg az utolsó fázis van folyamatban, kertépítés, parkosítás.</p>
    </div>
  </section>
)

export default Introduction