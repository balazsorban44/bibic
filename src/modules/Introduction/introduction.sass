.profiles
  display: grid
  grid-gap: 2em
  
.profile
  display: grid
  justify-items: center
  
  img
    margin-bottom: 1em
    width: 192px
    height: 192px
    border-radius: 50%

.history
  margin-top: 5% !important

@media (min-width: 768px)
  .history
    max-width: 540px
    margin: 0 auto
    p
      margin: 1em 0
    p::first-letter
      margin-left: 1em
      

  .profile
    h3
      grid-area: title
    a 
      grid-area: link
    h4
      grid-area: subtitle
    
    grid-gap: 0 2em
    grid-template-areas: "img ." "img title" "img link" "img subtitle" "img ."
    img
      grid-area: img
      margin-bottom: 0
      justify-self: end
    h3,h4,a
      justify-self: start

    &:first-of-type
      margin-left: -35vw
    &:last-of-type
      margin-left: -20vw
    &:nth-of-type(odd)
      h3,h4
        margin-left: -.25em
      
    &:nth-of-type(even)
      margin: -3.5vw -35vw -3.5vw 0
      h3,h4
        margin-right: -.25em
      h3,h4,a
        justify-self: end

      img
        justify-self: start
      grid-template-areas: ". img" "title img" "link img" "subtitle img" ". img"
      