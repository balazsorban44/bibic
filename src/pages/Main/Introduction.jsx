import React, {Suspense, lazy} from 'react'
import hegedus from 'assets/images/intro/hegedus.jpg'
import hegedusne from 'assets/images/intro/hegedusne.jpg'
import gombkoto from 'assets/images/intro/gombkoto.jpg'
import Zoom from "react-reveal/Zoom"
import Fade from "react-reveal/Fade"
import {Loading} from 'components/Elements'
import {useTranslation} from 'react-i18next'

const Paragraphs = lazy(() => import("components/Paragraphs"))

const profiles = [
  {
    name: {familyName: "Hegedüs", givenName: "Péter"},
    src: hegedus,
    email: "hegedus@bibicvendeghazak.hu",
    position: "host"
  },
  {
    name: {familyName: "Hegedüsné Kóró", givenName: "Ágnes"},
    src: hegedusne,
    email: "hegedusne@bibicvendeghazak.hu"
  },
  {
    name: {familyName: "Gombkötő", givenName: "Gábor"},
    src: gombkoto,
    email: "gombkoto@bibicvendeghazak.hu"
  }
]

export default () => {
  const [t] = useTranslation()
  return (
    <section id="bemutatkozas">
      <Fade>
        <h2 className="welcome-title">{t("introduction.title")}</h2>
      </Fade>
      <ul className="profiles">
        {profiles.map(({
          name, email, src, position = "owner"
        }) =>
          <Zoom
            duration={600}
            key={email}
          >
            <li >
              <a href={`mailto:${email}`}>
                <img
                  alt={t("name", name)}
                  className="profile-img"
                  src={src}
                />
              </a>
              <div>
                <h3>{t("name", name)}</h3>
                <h5><a href={`mailto:${email}`}>✉ {email} </a></h5>
                <h4>{t(`introduction.position.${position}`)}</h4>
              </div>
            </li>
          </Zoom>
        )}
      </ul>
      <div className="history">
        <Suspense fallback={<Loading/>}>
          <Paragraphs section="introduction"/>
        </Suspense>
      </div>
    </section>
  )
}
