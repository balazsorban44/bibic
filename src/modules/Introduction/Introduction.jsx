import React, {Suspense, lazy} from "react"
import {useTranslation} from "react-i18next"
import Zoom from "react-reveal/Zoom"


import hegedus from "assets/images/profiles/hegedus.jpg"
import hegedusne from "assets/images/profiles/hegedusne.jpg"
import gombkoto from "assets/images/profiles/gombkoto.jpg"

import Loading from "ui/Loading"
import Text from "ui/Text"
import Section from "ui/Section"

import "./introduction.sass"
import colors from "ui/utils/colors"

const Paragraphs = lazy(() => import("components/Paragraphs"))

const profiles = [
  {
    name: {
      family: "Hegedüs",
      given: "Péter"
    },
    src: hegedus,
    email: "hegedus@bibicvendeghazak.hu",
    role: "host"
  },
  {
    name: {
      family: "Hegedüsné Kóró",
      given: "Ágnes"
    },
    src: hegedusne,
    email: "hegedusne@bibicvendeghazak.hu",
    role: "owner"
  },
  {
    name: {
      family: "Gombkötő",
      given: "Gábor"
    },
    src: gombkoto,
    email: "gombkoto@bibicvendeghazak.hu",
    role: "owner"
  }
]


const Profile = ({t}) => ({name, email, src, role}) =>
  <Zoom duration={600} key={email}>
    <li className="profile">
      <img
        alt={name}
        className="profile-img"
        src={src}
      />
      <Text variant="h3">
        {t("name", {name})}
      </Text>
      <Text variant="h4">
        <a href={`mailto:${email}`}>✉ {email} </a>
      </Text>
      <Text variant="h5">
        {t(`roles.${role}`)}
      </Text>
    </li>
  </Zoom>


const Introduction = () => {
  const [t] = useTranslation("introduction")
  return (
    <Section
      bg="noisy"
      id="bemutatkozas"
      title={t("title")}
      titleProps={{style: {color: colors.accentColor2}}}
    >
      <ul className="profiles">
        {profiles.map(Profile({t}))}
      </ul>
      <Suspense fallback={<Loading/>}>
        <Paragraphs
          className="history"
          type="bemutatkozas"
        />
      </Suspense>
    </Section>
  )
}

export default Introduction