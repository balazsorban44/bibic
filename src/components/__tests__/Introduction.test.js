import Introduction from "../Introduction"

describe("Introduction component", () => {
  const wrapper = shallow(<Introduction/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

})