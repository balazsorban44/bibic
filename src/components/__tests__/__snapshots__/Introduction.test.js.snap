// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction component renders correctly 1`] = `ShallowWrapper {}`;
